openaddr/util/__init__.py

```diff
@@ -205,3 +210,118 @@ def s3_key_url(key):
     path = join(key.bucket.name, key.name.lstrip('/'))
     
     return urljoin(base, path)
+
+def get_cpu_times():
+    ''' Return Linux CPU usage times in jiffies.
+    
+        See http://stackoverflow.com/questions/1420426/how-to-calculate-the-cpu-usage-of-a-process-by-pid-in-linux-from-c
+    '''
+    if not exists('/proc/stat') or not exists('/proc/{}/stat'.format(getpid())):
+        return None, None, None
+    
+    with open('/proc/stat') as file:
+        stat = re.split(r'\s+', next(file).strip())
+        time_total = sum([int(s) for s in stat[1:]])
+
+    with open('/proc/{}/stat'.format(getpid())) as file:
+        stat = next(file).strip().split(' ')
+        utime, stime = (int(s) for s in stat[13:15])
+    
+    return time_total, utime, stime
+
+def get_diskio_bytes():
```

Comment:
this will measure all bytes read in the process, and so includes reading in shared libraries, etc; not just our productive data processing activity.